test/integration/basic/test/client-navigation.js

```diff
@@ -354,16 +354,16 @@ export default (context, render) => {
         })
       })
 
-      describe('when hash get removed', () => {
+      describe('when hash gets removed', () => {
         it('should not run getInitialProps', async () => {
           const browser = await webdriver(context.appPort, '/nav/hash-changes')
 
           const counter = await browser
-            .elementByCss('#via-a').click()
+            .elementByCss('#via-link').click()
             .elementByCss('#page-url').click()
             .elementByCss('p').text()
 
-          expect(counter).toBe('COUNT: 1')
+          expect(counter).toBe('COUNT: 0')
```

Comment:
This test wasn't actually testing what it described; if `getInitialProps` was not called, count should be zero.

As far as I can tell, the Next.js router doesn't handle regular anchor-based hash changes outside of the Link component (e.g. clicking an anchor hash doesn't update the router state, so clicking #page-url in that case WOULD trigger getInitialProps)